src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Queries/GameResultQueries.cs

```diff
@@ -69,25 +69,45 @@ public GameResultDto Execute(FindByIdCriteria criteria)
         /// <returns>List of domain models of game result.</returns>
         public List<GameResultDto> Execute(TournamentGameResultsCriteria criteria)
         {
-            var query =
-                (from game in _dalGameResults
-                 join tournament in _dalTournaments
-                     on game.TournamentId equals tournament.Id
-                 join division in _dalDivisions
-                     on tournament.Id equals division.TournamentId
-                 join groups in _dalGroups
-                     on division.Id equals groups.DivisionId
-                 where game.TournamentId == criteria.TournamentId
-                 select new
-                 {
-                     results = game,
-                     divisionName = division.Name,
-                     divisionId = division.Id,
-                     groupId = groups.Id
-                 })
-                // Distinct by game id
-                .GroupBy(p => p.results.Id)
-                .Select(group => group.FirstOrDefault());
+            var tournamentId = criteria.TournamentId;
+
+            var allGamesWithTeams =
+                from game in _dalGameResults
+                join tournament in _dalTournaments
+                    on game.TournamentId equals tournament.Id
+                join division in _dalDivisions
+                    on tournament.Id equals division.TournamentId
+                join groups in _dalGroups
+                    on division.Id equals groups.DivisionId
+                where game.TournamentId == tournamentId
+                where groups.Teams.Contains(game.HomeTeam)
+                select new
+                {
+                    results = game,
+                    divisionName = division.Name,
+                    divisionId = division.Id,
+                    groupId = groups.Id
+                };
+
+            var gamesWithoutTeams =
+                from game in _dalGameResults
+                join tournament in _dalTournaments
+                    on game.TournamentId equals tournament.Id
+                join division in _dalDivisions
+                    on tournament.Id equals division.TournamentId
+                join groups in _dalGroups
+                    on division.Id equals groups.DivisionId
+                where game.TournamentId == tournamentId
+                where game.HomeTeam == null
+                select new
+                {
+                    results = game,
+                    divisionName = division.Name,
+                    divisionId = division.Id,
+                    groupId = groups.Id
+                };
+
+            var query = allGamesWithTeams.Union(gamesWithoutTeams);
```

Comment:
`-- Region Parameters
DECLARE @p__linq__0 Int = 13
DECLARE @p__linq__1 Int = 13
-- EndRegion
SELECT 
    [Distinct1].[C1] AS [C1], 
    [Distinct1].[C2] AS [C2], 
    [Distinct1].[C3] AS [C3], 
    [Distinct1].[C4] AS [C4], 
    [Distinct1].[C5] AS [C5], 
    [Distinct1].[C6] AS [C6], 
    [Distinct1].[C7] AS [C7], 
    [Distinct1].[C8] AS [C8], 
    [Distinct1].[C9] AS [C9], 
    [Distinct1].[C10] AS [C10], 
    [Distinct1].[C11] AS [C11], 
    [Distinct1].[C12] AS [C12], 
    [Distinct1].[C13] AS [C13], 
    [Distinct1].[C14] AS [C14], 
    [Distinct1].[C15] AS [C15], 
    [Distinct1].[C16] AS [C16], 
    [Distinct1].[C17] AS [C17], 
    [Distinct1].[C18] AS [C18], 
    [Distinct1].[C19] AS [C19], 
    [Distinct1].[C20] AS [C20], 
    [Distinct1].[C21] AS [C21], 
    [Distinct1].[C22] AS [C22], 
    [Distinct1].[C23] AS [C23], 
    [Distinct1].[C24] AS [C24], 
    [Distinct1].[C25] AS [C25], 
    [Distinct1].[C26] AS [C26], 
    [Distinct1].[C27] AS [C27], 
    [Distinct1].[C28] AS [C28], 
    [Distinct1].[C29] AS [C29], 
    [Distinct1].[C30] AS [C30], 
    [Distinct1].[C32] AS [C31], 
    [Distinct1].[C33] AS [C32], 
    [Distinct1].[C34] AS [C33]
    FROM ( SELECT DISTINCT 
        [UnionAll1].[C1] AS [C1], 
        [UnionAll1].[Id] AS [C2], 
        [UnionAll1].[TournamentId] AS [C3], 
        [UnionAll1].[HomeTeamId] AS [C4], 
        [UnionAll1].[AwayTeamId] AS [C5], 
        [UnionAll1].[HomeSetsScore] AS [C6], 
        [UnionAll1].[AwaySetsScore] AS [C7], 
        [UnionAll1].[IsTechnicalDefeat] AS [C8], 
        [UnionAll1].[HomeSet1Score] AS [C9], 
        [UnionAll1].[AwaySet1Score] AS [C10], 
        [UnionAll1].[IsSet1TechnicalDefeat] AS [C11], 
        [UnionAll1].[HomeSet2Score] AS [C12], 
        [UnionAll1].[AwaySet2Score] AS [C13], 
        [UnionAll1].[IsSet2TechnicalDefeat] AS [C14], 
        [UnionAll1].[HomeSet3Score] AS [C15], 
        [UnionAll1].[AwaySet3Score] AS [C16], 
        [UnionAll1].[IsSet3TechnicalDefeat] AS [C17], 
        [UnionAll1].[HomeSet4Score] AS [C18], 
        [UnionAll1].[AwaySet4Score] AS [C19], 
        [UnionAll1].[IsSet4TechnicalDefeat] AS [C20], 
        [UnionAll1].[HomeSet5Score] AS [C21], 
        [UnionAll1].[AwaySet5Score] AS [C22], 
        [UnionAll1].[IsSet5TechnicalDefeat] AS [C23], 
        [UnionAll1].[StartTime] AS [C24], 
        [UnionAll1].[RoundNumber] AS [C25], 
        [UnionAll1].[GameNumber] AS [C26], 
        [UnionAll1].[PenaltyTeam] AS [C27], 
        [UnionAll1].[PenaltyAmount] AS [C28], 
        [UnionAll1].[PenaltyDescription] AS [C29], 
        [UnionAll1].[UrlToGameVideo] AS [C30], 
        [UnionAll1].[HomeTeamId1] AS [C31], 
        [UnionAll1].[Name] AS [C32], 
        [UnionAll1].[Id1] AS [C33], 
        [UnionAll1].[Id2] AS [C34]
        FROM  (SELECT 
            1 AS [C1], 
            [Extent1].[Id] AS [Id], 
            [Extent1].[TournamentId] AS [TournamentId], 
            [Extent1].[HomeTeamId] AS [HomeTeamId], 
            [Extent1].[AwayTeamId] AS [AwayTeamId], 
            [Extent1].[HomeSetsScore] AS [HomeSetsScore], 
            [Extent1].[AwaySetsScore] AS [AwaySetsScore], 
            [Extent1].[IsTechnicalDefeat] AS [IsTechnicalDefeat], 
            [Extent1].[HomeSet1Score] AS [HomeSet1Score], 
            [Extent1].[AwaySet1Score] AS [AwaySet1Score], 
            [Extent1].[IsSet1TechnicalDefeat] AS [IsSet1TechnicalDefeat], 
            [Extent1].[HomeSet2Score] AS [HomeSet2Score], 
            [Extent1].[AwaySet2Score] AS [AwaySet2Score], 
            [Extent1].[IsSet2TechnicalDefeat] AS [IsSet2TechnicalDefeat], 
            [Extent1].[HomeSet3Score] AS [HomeSet3Score], 
            [Extent1].[AwaySet3Score] AS [AwaySet3Score], 
            [Extent1].[IsSet3TechnicalDefeat] AS [IsSet3TechnicalDefeat], 
            [Extent1].[HomeSet4Score] AS [HomeSet4Score], 
            [Extent1].[AwaySet4Score] AS [AwaySet4Score], 
            [Extent1].[IsSet4TechnicalDefeat] AS [IsSet4TechnicalDefeat], 
            [Extent1].[HomeSet5Score] AS [HomeSet5Score], 
            [Extent1].[AwaySet5Score] AS [AwaySet5Score], 
            [Extent1].[IsSet5TechnicalDefeat] AS [IsSet5TechnicalDefeat], 
            [Extent1].[StartTime] AS [StartTime], 
            [Extent1].[RoundNumber] AS [RoundNumber], 
            [Extent1].[GameNumber] AS [GameNumber], 
            [Extent1].[PenaltyTeam] AS [PenaltyTeam], 
            [Extent1].[PenaltyAmount] AS [PenaltyAmount], 
            [Extent1].[PenaltyDescription] AS [PenaltyDescription], 
            [Extent1].[UrlToGameVideo] AS [UrlToGameVideo], 
            [Extent1].[HomeTeamId] AS [HomeTeamId1], 
            [Extent2].[Name] AS [Name], 
            [Extent2].[Id] AS [Id1], 
            [Extent3].[Id] AS [Id2]
            FROM   [dbo].[GameResults] AS [Extent1]
            INNER JOIN [dbo].[Divisions] AS [Extent2] ON [Extent1].[TournamentId] = [Extent2].[TournamentId]
            INNER JOIN [dbo].[Groups] AS [Extent3] ON [Extent2].[Id] = [Extent3].[DivisionId]
            WHERE ([Extent1].[TournamentId] = @p__linq__0) AND ( EXISTS (SELECT 
                1 AS [C1]
                FROM [dbo].[GroupTeam] AS [Extent4]
                WHERE ([Extent1].[HomeTeamId] = [Extent4].[TeamId]) AND ([Extent3].[Id] = [Extent4].[GroupId])
            ))
        UNION ALL
            SELECT 
            1 AS [C1], 
            [Extent5].[Id] AS [Id], 
            [Extent5].[TournamentId] AS [TournamentId], 
            [Extent5].[HomeTeamId] AS [HomeTeamId], 
            [Extent5].[AwayTeamId] AS [AwayTeamId], 
            [Extent5].[HomeSetsScore] AS [HomeSetsScore], 
            [Extent5].[AwaySetsScore] AS [AwaySetsScore], 
            [Extent5].[IsTechnicalDefeat] AS [IsTechnicalDefeat], 
            [Extent5].[HomeSet1Score] AS [HomeSet1Score], 
            [Extent5].[AwaySet1Score] AS [AwaySet1Score], 
            [Extent5].[IsSet1TechnicalDefeat] AS [IsSet1TechnicalDefeat], 
            [Extent5].[HomeSet2Score] AS [HomeSet2Score], 
            [Extent5].[AwaySet2Score] AS [AwaySet2Score], 
            [Extent5].[IsSet2TechnicalDefeat] AS [IsSet2TechnicalDefeat], 
            [Extent5].[HomeSet3Score] AS [HomeSet3Score], 
            [Extent5].[AwaySet3Score] AS [AwaySet3Score], 
            [Extent5].[IsSet3TechnicalDefeat] AS [IsSet3TechnicalDefeat], 
            [Extent5].[HomeSet4Score] AS [HomeSet4Score], 
            [Extent5].[AwaySet4Score] AS [AwaySet4Score], 
            [Extent5].[IsSet4TechnicalDefeat] AS [IsSet4TechnicalDefeat], 
            [Extent5].[HomeSet5Score] AS [HomeSet5Score], 
            [Extent5].[AwaySet5Score] AS [AwaySet5Score], 
            [Extent5].[IsSet5TechnicalDefeat] AS [IsSet5TechnicalDefeat], 
            [Extent5].[StartTime] AS [StartTime], 
            [Extent5].[RoundNumber] AS [RoundNumber], 
            [Extent5].[GameNumber] AS [GameNumber], 
            [Extent5].[PenaltyTeam] AS [PenaltyTeam], 
            [Extent5].[PenaltyAmount] AS [PenaltyAmount], 
            [Extent5].[PenaltyDescription] AS [PenaltyDescription], 
            [Extent5].[UrlToGameVideo] AS [UrlToGameVideo], 
            [Extent5].[HomeTeamId] AS [HomeTeamId1], 
            [Extent6].[Name] AS [Name], 
            [Extent6].[Id] AS [Id1], 
            [Extent7].[Id] AS [Id2]
            FROM   [dbo].[GameResults] AS [Extent5]
            INNER JOIN [dbo].[Divisions] AS [Extent6] ON [Extent5].[TournamentId] = [Extent6].[TournamentId]
            INNER JOIN [dbo].[Groups] AS [Extent7] ON [Extent6].[Id] = [Extent7].[DivisionId]
            WHERE ([Extent5].[HomeTeamId] IS NULL) AND ([Extent5].[TournamentId] = @p__linq__1)) AS [UnionAll1]
    )  AS [Distinct1]
`